day04/lib/main.rb

```diff
@@ -0,0 +1,114 @@
+# => TODO
+# => warehouse (how many product we have avaible)
+# => total price in basket + tax
+
+class Main
+  require_relative "./product"
+  require_relative "./warehouse"
+  require_relative "./basket"
```

Comment:
These `require_relative` lines should go at the top of the file, outside of class.